### ISSUE TITLE
Substitutions in External File Flagged as Error

### ISSUE THREAD

**Issue by markperfectsensedigital** [twolfson/restructuredtext-lint#39]
If an rST project has substitutions in an external file, restructuredtext-lint flags substitutions in the current file as an error. For example, in the attached illustration, the file `installing.txt` has some substitutions in an external file `substitutions.tsr`. The sphinx build is clean, so this substitution really isn't an error. However, the linter reports `[restructuredtext-lint] Undefined substitution referenced: "emdash".`

![rst-linter](https://user-images.githubusercontent.com/24228071/28573811-dd193970-7119-11e7-9ddb-a44f7abd2281.png)

Possible resolutions:

* Linter checks if a substitution is defined somewhere in the rST project
* Include an option to disable checks for undefined substitutions.

Thank you.

**Comment by twolfson** [comment]
Ah, I don't think I've encountered substitutions before. Can you provide how we load them into `installing.txt`?

**Comment by markperfectsensedigital** [comment]
Generally, rST substitutions are a macro. Anything within |two pipes| is expanded to the substitution.  

At our company we place common substitution definitions in a single file, and use the rst_prolog option to include that file at the beginning of every rST file.

In the attached project:

- substitions.tsr holds the substitution definition.
- body.rst, line 6, uses the substitution.
- conf.py, line 171, uses the rst_prolog configuration to place substitutions.tsr at the beginning of body.rst

The result is in _build/html/body.html

[lint_test.zip](https://github.com/twolfson/restructuredtext-lint/files/1174793/lint_test.zip)

**Comment by twolfson** [comment]
Ah, neat. Thanks for all the info!

My gut reaction is that we won't be able to support this in the format done in the project. It would rely too heavily on a Sphinx project configuration where we've taken a stance of not supporting Sphinx fully due to directive issues:

https://github.com/twolfson/restructuredtext-lint/tree/1.0.1#sphinx

I think we'd be open to supporting `rst_prolog` in some capacity (i.e. CLI flag for file or content itself, offset lines for reported errors). For example:

```bash
rst-lint README.rst --rst-prolog "
.. include:: substitutions.tsr
"
```

What are your thoughts?

**Comment by markperfectsensedigital** [comment]
Sounds good to me! Thank you.

**Comment by twolfson** [comment]
Alright, starting on implementing this now

**Comment by twolfson** [comment]
`--rst-prolog` support has been landed and released in `1.1.0`. Thanks for the issue and quick responses
